packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,193 @@
+import * as React from 'react';
+import { KeyCodes } from '../../Utilities';
+import { TimeConstants, addMinutes, formatTimeString, ceilMinuteToIncrement } from '@fluentui/date-time-utilities';
+import { ComboBox, IComboBox, IComboBoxOption } from '../../ComboBox';
+import { ITimePickerProps, ITimeRange, ITimePickerStrings } from './TimePicker.types';
+
+const REGEX_SHOW_SECONDS_HOUR_12 = /((1[0-2]|0?[1-9]):([0-5][0-9]):(?:[0-5]\d) ?([AaPp][Mm]))$/;
+const REGEX_HIDE_SECONDS_HOUR_12 = /((1[0-2]|0?[1-9]):([0-5][0-9]) ?([AaPp][Mm]))$/;
+const REGEX_SHOW_SECONDS_HOUR_24 = /([0-9]|0[0-9]|1[0-9]|2[0-3]):(?:[0-5]\d):(?:[0-5]\d)$/;
+const REGEX_HIDE_SECONDS_HOUR_24 = /([0-9]|0[0-9]|1[0-9]|2[0-3]):(?:[0-5]\d)$/;
+
+const TIME_LOWER_BOUND = 0;
+const TIME_UPPER_BOUND = 23;
+
+const getDefaultStrings = (useHour12: boolean, showSeconds: boolean): ITimePickerStrings => {
+  let errorMessageToDisplay = '';
+  const hourUnits = useHour12 ? '12-hour' : '24-hour';
+  showSeconds
+    ? (errorMessageToDisplay = `TimePicker format must be valid and in the ${hourUnits} ` + `format hh:mm:ss A.`)
+    : (errorMessageToDisplay = `TimePicker format must be valid and in the ${hourUnits} ` + `format hh:mm A.`);
+
+  return {
+    invalidInputErrorMessage: errorMessageToDisplay,
+  };
+};
+
+export const TimePicker: React.FunctionComponent<ITimePickerProps> = ({
```

Comment:
Something to consider for the future: right now it looks like the time picker is always completely uncontrolled (and doesn't even support passing a default value), but for some scenarios the consumer may want to pass a default value or manually control the value. 

In the interest of not complicating this PR even more, I think it's fine to wait on that feature until somebody requests it. But if/when you get around to implementing it, you'll need to ensure it implements the controlled/uncontrolled pattern properly--[here's an explanation I wrote](https://gist.github.com/ecraig12345/1ea4566ce171b844a0af5a3366d3473f), or you can find many other explanations online if that doesn't make sense. We also have a hook `useControllableValue` to help with implementing it properly.

Reply:
Will defer for the future 😝